great_expectations/datasource/fluent/config_str.py

```diff
@@ -53,7 +53,7 @@ def __repr__(self) -> str:
 
     @classmethod
     def _validate_template_str_format(cls, v):
-        if TEMPLATE_STR_REGEX.match(v):
+        if TEMPLATE_STR_REGEX.search(v):
```

Comment:
The problem was the use of `re.match` which only matches the start of the text instead of looking for a match anywhere in the text.

https://docs.python.org/3/library/re.html#re.search

>re.match() will only match at the beginning of the string